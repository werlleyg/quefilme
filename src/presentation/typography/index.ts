export * from "./H3";
export * from "./H4";
export * from "./P";
