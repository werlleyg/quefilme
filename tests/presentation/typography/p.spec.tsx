import { fireEvent, render, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import { P } from "@/presentation";
import React from "react";

describe("P Typography Component", () => {
  test("Should render the P component with children", () => {
    const { getByText } = render(<P>Test Content</P>);
    expect(getByText("Test Content")).toBeInTheDocument();
  });

  test("Should apply HTML attributes to the Paragraph element", () => {
    const { getByText } = render(
      <P id="test-id" className="test-class">
        Test Content
      </P>,
    );
    const element = getByText("Test Content");

    expect(element).toHaveAttribute("id", "test-id");
    expect(element).toHaveClass("test-class");
  });

  test("Should render without children", () => {
    const { container } = render(<P />);
    expect(container.firstChild).toBeInTheDocument();
    expect(container.firstChild).toBeEmptyDOMElement();
  });

  test("Should apply inline styles to the Paragraph element", () => {
    const { getByText } = render(<P style={{ color: "red" }}>Styled Text</P>);
    const element = getByText("Styled Text");

    expect(element).toHaveStyle("color: red");
  });

  test("Should call onClick handler when clicked", () => {
    const handleClick = jest.fn();
    const { getByText } = render(<P onClick={handleClick}>Clickable Text</P>);

    fireEvent.click(getByText("Clickable Text"));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  test("Should render children components correctly", () => {
    const ChildComponent = () => <span>Child Component</span>;

    const { getByText } = render(
      <P>
        <ChildComponent />
      </P>,
    );

    expect(getByText("Child Component")).toBeInTheDocument();
  });

  test("Should match the snapshot", () => {
    const { asFragment } = render(
      <P className="snapshot-test">Snapshot Content</P>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
