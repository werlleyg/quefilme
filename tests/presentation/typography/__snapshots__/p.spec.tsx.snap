// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`P Typography Component Should match the snapshot 1`] = `
<DocumentFragment>
  <p
    class="snapshot-test css-1watjdo"
  >
    Snapshot Content
  </p>
</DocumentFragment>
`;
